hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/BucketIndexUtil.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *
+ *  * Licensed to the Apache Software Foundation (ASF) under one
+ *  * or more contributor license agreements.  See the NOTICE file
+ *  * distributed with this work for additional information
+ *  * regarding copyright ownership.  The ASF licenses this file
+ *  * to you under the Apache License, Version 2.0 (the
+ *  * "License"); you may not use this file except in compliance
+ *  * with the License.  You may obtain a copy of the License at
+ *  *
+ *  *      http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package org.apache.hudi.sink.utils;
+
+import org.apache.hudi.index.bucket.BucketIdentifier;
+
+public class BucketIndexUtil {
+
+  /**
+   * This method is used to get the partition index of a bucket.
+   * "partition.hashCode() / (parallelism / bucketNum) * bucketNum" divides the parallelism into
+   * sub-intervals of length bucket_num, different partitions will be mapped to different sub-interval,
+   * ensure that the data across multiple partitions is evenly distributed.
+   * @param partition partition string
+   * @param bucketNum bucket number per partition
+   * @param parallelism parallelism of the task
+   * @return The partition index of this bucket.
+   */
+  public static int getPartitionIndex(String partition, int curBucket, int bucketNum, int parallelism) {
+    int partitionIndex;
+    if (parallelism < bucketNum) {
```

Comment:
Can we add a boolean variable to the `BucketIndexPartitioner`, because we don't need to perform conditional judgment on every record?